arro3-core/python/arro3/core/types.py

```diff
@@ -3,14 +3,14 @@
 import array as _array
 import mmap
 import sys
-from typing import TYPE_CHECKING, Protocol, Tuple, Union
-
-if sys.version_info >= (3, 12):
-    from collections.abc import Buffer as _Buffer
-else:
-    from typing_extensions import Buffer as _Buffer
+from typing import TYPE_CHECKING, Any, Protocol, Tuple, Union
 
 if TYPE_CHECKING:
+    if sys.version_info >= (3, 12):
+        from collections.abc import Buffer as _Buffer
+    else:
+        from typing_extensions import Buffer as _Buffer
```

Comment:
This move is fine but I think we want to add `typing-extensions` into our dependency list for `<3.12` anyways.